hedera-node/src/main/java/com/hedera/services/txns/consensus/TopicCreateTransitionLogic.java

```diff
@@ -21,113 +21,82 @@
 import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.BAD_ENCODING;
 import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_AUTORENEW_ACCOUNT;
 import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_RENEWAL_PERIOD;
 import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
-import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SUCCESS;
 
 /**
  * The syntax check pre-consensus validates the adminKey's structure as signature validation occurs before
  * doStateTransition().
  */
 @Singleton
 public class TopicCreateTransitionLogic implements TransitionLogic {
-	private static final Logger log = LogManager.getLogger(TopicCreateTransitionLogic.class);
-
-	private final Function<TransactionBody, ResponseCodeEnum> PRE_SIGNATURE_VALIDATION_SEMANTIC_CHECK =
-			this::validatePreSignatureValidation;
-
-	private final HederaLedger ledger;
-	private final Supplier<MerkleMap<EntityNum, MerkleAccount>> accounts;
-	private final Supplier<MerkleMap<EntityNum, MerkleTopic>> topics;
+	private final AccountStore accountStore;
+	private final TopicStore topicStore;
 	private final EntityIdSource entityIdSource;
 	private final OptionValidator validator;
+	private final Function<TransactionBody, ResponseCodeEnum> PRE_SIGNATURE_VALIDATION_SEMANTIC_CHECK = this::validate;
 	private final TransactionContext transactionContext;
 
 	@Inject
 	public TopicCreateTransitionLogic(
-			Supplier<MerkleMap<EntityNum, MerkleAccount>> accounts,
-			Supplier<MerkleMap<EntityNum, MerkleTopic>> topics,
-			EntityIdSource entityIdSource,
-			OptionValidator validator,
-			TransactionContext transactionContext,
-			HederaLedger ledger
-	) {
-		this.accounts = accounts;
-		this.topics = topics;
-		this.ledger = ledger;
+			final TopicStore topicStore,
+			final EntityIdSource entityIdSource,
+			final OptionValidator validator,
+			final TransactionContext transactionContext,
+			final AccountStore accountStore) {
+		this.accountStore = accountStore;
+		this.topicStore = topicStore;
 		this.entityIdSource = entityIdSource;
 		this.validator = validator;
 		this.transactionContext = transactionContext;
 	}
 
 	@Override
 	public void doStateTransition() {
-		var postConsensusValidationResult = validatePreStateTransition();
-		if (OK != postConsensusValidationResult) {
-			transactionContext.setStatus(postConsensusValidationResult);
-			return;
+		/* --- Extract gRPC --- */
+		final var transactionBody = transactionContext.accessor().getTxn();
+		final var payerAccountId = transactionBody.getTransactionID().getAccountID();
+		final var op = transactionBody.getConsensusCreateTopic();
+		/* --- Validate --- */
+		final var memoValidationResult = validator.memoCheck(op.getMemo());
+		validateTrue(OK == memoValidationResult, memoValidationResult);
+		validateFalse(op.hasSubmitKey() && !validator.hasGoodEncoding(op.getSubmitKey()), BAD_ENCODING);
```

Comment:
We do not need to have this check here. If the submit key is not valid we will throw in the `Topic.createFromGrpc` method.